File: apps/backend/internal/users/domain/entity/companion.go
package entity

// CompanionProfile holds the public companion profile shown to clients.
type CompanionProfile struct {
	CompanionID     string   `json:"companion_id"`
	Rating          float64  `json:"rating"`
	ExperienceYears int      `json:"experience_years"`
	Languages       []string `json:"languages"`
	Services        []string `json:"services"`
	Bio             string   `json:"bio,omitempty"`
}

// Companion is the read model returned by GET /companions:
// a user with role "companion" joined with its public profile.
type Companion struct {
	ID              string   `json:"id"`
	Email           string   `json:"email"`
	FirstName       string   `json:"first_name"`
	LastName        string   `json:"last_name"`
	Phone           string   `json:"phone,omitempty"`
	Avatar          string   `json:"avatar,omitempty"`
	Rating          float64  `json:"rating"`
	ExperienceYears int      `json:"experience_years"`
	Languages       []string `json:"languages"`
	Services        []string `json:"services"`
	Bio             string   `json:"bio,omitempty"`
}

func (c *Companion) FullName() string {
	return c.FirstName + " " + c.LastName
}

File: apps/legacy/mobile-flutter/lib/features/companion/domain/entities/companion.dart
class Companion {
  final String id;
  final String email;
  final String firstName;
  final String lastName;
  final String? phone;
  final String? avatar;
  final double rating;
  final int experienceYears;
  final List<String> languages;
  final List<String> services;
  final String? bio;

  const Companion({
    required this.id,
    required this.email,
    required this.firstName,
    required this.lastName,
    this.phone,
    this.avatar,
    this.rating = 5.0,
    this.experienceYears = 0,
    this.languages = const [],
    this.services = const [],
    this.bio,
  });

  String get fullName => '$firstName $lastName';

  String get experienceLabel {
    if (experienceYears <= 0) return 'Nuevo';
    if (experienceYears == 1) return '1 año de experiencia';
    return '$experienceYears años de experiencia';
  }

  bool supportsService(String serviceType) => services.isEmpty || services.contains(serviceType);

  bool matchesQuery(String query) {
    final q = query.trim().toLowerCase();
    if (q.isEmpty) return true;
    return fullName.toLowerCase().contains(q) ||
        languages.any((l) => l.toLowerCase().contains(q)) ||
        services.any((s) => s.toLowerCase().contains(q));
  }
}
